jamaljsr/polar
# feat: add custom baseport for nodes

This commit implements the feature that lets users set custom base ports for their nodes to avoid having port conflicts with other applications running on the same machine.

Closes #533 

### Description

I added a settings page with input fields to set the base ports and then save. The custom base ports persist in the settings.json file and are loaded whenever polar starts. If a node, for example, `LND` does not have a custom port set, the defaults from [constant.ts](https://github.com/jamaljsr/polar/blob/master/src/utils/constants.ts) are used

### Steps to Test

1. On the top navbar, locate the `Network Setting` tab and click on it.
2. The settings page appears, fill in the custom ports and click on save.
3. Create a network and check the ports of your nodes

### Screenshots
![Screenshot from 2024-04-23 18-07-58](https://github.com/jamaljsr/polar/assets/45062451/7c61fbb9-ce38-4efc-90f4-5a572ff62a72)

![Screenshot from 2024-04-23 18-09-15](https://github.com/jamaljsr/polar/assets/45062451/a8e1f9cd-3d0c-4cb6-b292-6e1cfa213b30)

Node Alice
![Screenshot from 2024-04-23 18-10-45](https://github.com/jamaljsr/polar/assets/45062451/c6829899-0b3e-4e9d-90c8-9154c662d0b9)

Node Bob
![Screenshot from 2024-04-23 18-11-05](https://github.com/jamaljsr/polar/assets/45062451/0258974a-5e42-4681-b20a-8c50e4998668)
